src/devices/cddasongloader.cpp

```diff
@@ -122,9 +123,12 @@ void CddaSongLoader::LoadSongsFromCdda() {
   GstMessage* msg = nullptr;
   GstMessage* msg_toc = nullptr;
   GstMessage* msg_tag = nullptr;
-  while ((msg = gst_bus_timed_pop_filtered(
-              GST_ELEMENT_BUS(pipeline), 2 * GST_SECOND,
-              (GstMessageType)(GST_MESSAGE_TOC | GST_MESSAGE_TAG)))) {
+  GstMessageType msg_filter =
+      static_cast<GstMessageType>(GST_MESSAGE_TOC | GST_MESSAGE_TAG);
+  while ((msg = gst_bus_timed_pop_filtered(GST_ELEMENT_BUS(pipeline),
+                                           10 * GST_SECOND, msg_filter))) {
+    msg_filter = static_cast<GstMessageType>(
+        static_cast<int>(msg_filter) ^ static_cast<int>(GST_MESSAGE_TYPE(msg)));
```

Comment:
According to the docs, we may receive multiple TAG messages with different pieces of data. Maybe we should parse the message in the loop and filter the message after we find the disc id?
https://gstreamer.freedesktop.org/documentation/application-development/advanced/metadata.html

Reply:
Good point, thanks. Not sure how to handle this best - I think I'll look into setting up a callback that waits for a tag message containing the discid.